crates/storage/provider/src/test_utils/mock.rs

```diff
@@ -244,3 +244,23 @@ impl StateProviderFactory for MockEthProvider {
         todo!()
     }
 }
+
+impl StateProviderFactory for Arc<MockEthProvider> {
+    type HistorySP<'a> = &'a MockEthProvider where Self: 'a;
+    type LatestSP<'a> = &'a MockEthProvider where Self: 'a;
+
+    fn latest(&self) -> Result<Self::LatestSP<'_>> {
```

Comment:
this _feels_ a bit weird, but let's see

Reply:
ideally, i'd like to refactor StateProvider to be able to auto impl `Arc`, but, at the very least we should change HistorySP and LatestSP here to some wrapper type of references to MockEthProvider inner collections